tightenco/ziggy
# ES6, Webpack and much more

This PR is based on #86 

### Highlights
* ES6 everywhere
* Webpack 
    - Minimal setup 
    - Generate two dist files, minified and non-minified
    - [babel-preset-env](https://babeljs.io/env/) and [browserlist](http://browserl.ist/?q=%3E+2%25%2C+last+2+versions) take care of browser support
   - Fixes #79 IE issue
   -  Watch mode for continuous build
   - `npm run build:watch`
* Mocha-webpack
   - Test case can use ES6 syntax
   - Build (run webppack) then test against `dist` files
   - Watch mode to see test results in realtime
   - `npm run test:watch`
* Specify node js version in package.json file
     - This will prevent `npm install` on older version and force same version across contributors machines
* Throw Error object `throw new Error()` instead of string
* Update .gitignore
     - ignore phpStorm/Webstorm generated files
     - ignore any *.log file (usually created by node)
* API remains the same
   - Users can still use `route()` method same as before
   - We are taking advantage of webpack [library](https://webpack.js.org/guides/author-libraries) build to expose `route` method to window object
  - Dist files are ready to use in module environment, like
  - `import route from 'dist/js/route.js'`
  - Dist files can be dumped to html page and works without any module bundler too
 * Speed up CI by caching vendor files

@DanielCoulbourne 
Feel free to checkout this PR [locally](https://help.github.com/articles/checking-out-pull-requests-locally/) and make your changes.
